### ISSUE TITLE
Диплом

### ISSUE THREAD

**Issue by PtashkaStep** [dronpavlove/python_basic_diploma#1]
Привет, тоже делаю этого бота и возникли сложности. Не мог бы помочь?

**Comment by dronpavlove** [comment]
Привет! Попробую, а в чем вопрос?

**Comment by PtashkaStep** [comment]
не мог ли бы вы, скинуть файлы которые у вас в гитигнор?

**Comment by dronpavlove** [comment]
У меня их не осталось... Но тут Вам важен только settings.py
В нём вот так должно быть:
TOKEN = "тут токен который выдаст телеграмм при регистрации бота"
# а эти данные даст сайт, на которые запросы отправлять
headers_dict = {
        'x-rapidapi-key': 'ee3675b80emsh30b7faec807b937p137727jsn27ea13765b80',
        'x-rapidapi-host': 'hotels4.p.rapidapi.com'
    }

**Comment by PtashkaStep** [comment]
ну это есть, просто бот при вводе текста в телеграме, выдает ошибку пустой текст

**Comment by PtashkaStep** [comment]
бот отвечает на команды у меня, которые через /start и прочие такие команды, но когда ввожу город, просто не реагирует

**Comment by dronpavlove** [comment]
Не появляется "веду поиск, надо подождать"? Если появляется, значит проблема с подключением к сайту гостиниц. Или инет слабый, или у них там бывало такое- тех.работы. А ещё могло количество запросов более 500 уже у Вас быть, тогда надо по новой ключи к API сайта просить с другого емейла... В любом случае, если проблема в коде- он ошибку в терминал должен написать (или в файл exeption,  уже не помню) Ну, если завтра к вечеру не разберётесь, попробую подключить его к работе...

**Comment by PtashkaStep** [comment]
веду поиск, надо подождать, не выходит даже

**Comment by PtashkaStep** [comment]
хорошо, спасибо

**Comment by PtashkaStep** [comment]
database.json какие данные в этом файле были? как  я понимаю там дынные городов и отелей?

**Comment by dronpavlove** [comment]
Да нет, там я хранил просто базу пользователей, которые пользовались ботом. Я сейчас на работе, вечером посмотрю

**Comment by dronpavlove** [comment]
Добрый вечер! Посмотрите мой новы файл proba.py Там я вижу, что поменялись в АПИ возвращаемые после запросов данные. Надо код менять... Может мои прикидки в этом файле Вам помогут.

**Comment by PtashkaStep** [comment]
спасибо, сейчас попробую

**Comment by averageprogenjoyer** [comment]
Присоединюсь к обсуждению, тоже делаю этот проект, с файлом proba.py все хорошо, он получает информацию по API, но как я понял, class_for_bot.py не обрабатывает и не выводит эту информацию (если, конечно, я правильно понял) 
upd. Пересмотрел, информацию обрабатывает в нужный список, а сам список почему-то не выводит..

**Comment by dronpavlove** [comment]
Добрый день! Да, я поэтому и кинул proba.py У меня в логике обработки результата (когда я защищался) всё работало. Сейчас api отдаёт другой ответ, не такой, который был. И теперь логику запросов и их обработки надо менять. В файле proba.py я пару дней назад просто прикинул, что возвращается и как это можно обработать.

**Comment by averageprogenjoyer** [comment]
А можете подсказать, какая часть кода отвечает за отправку информации об отелях после этого
![изображение](https://user-images.githubusercontent.com/102250954/227244550-2914e496-eea4-48df-bc07-b96746d04946.png)

**Comment by dronpavlove** [comment]
Я там новым комитом поправочки внёс, но это наброски. @MyFirstTry_tele_bot мой бот отдаёт теперь сразу полный список отелей с кнопками. Надо теперь обработчику кнопок "объяснять", как действовать дальше...

**Comment by averageprogenjoyer** [comment]
Тоесть, если я теперь скачаю весь проект - вывод отелей будет работать?
Просто, делаю исправления как в коммите - результат тот же, ведется поиск в городе москва, и все

**Comment by dronpavlove** [comment]
Нет, выводится список совпавших городов. Теперь надо запрос поиска отелей поправлять. В api искать, каким запросом получить отели, зная destinationId города

**Comment by dronpavlove** [comment]
Но вот этот запрос у них сейчас не отдает ничего почему-то...
import requests

url = "https://hotels4.p.rapidapi.com/properties/list"

querystring = {"destinationId":"1506246","pageNumber":"1","pageSize":"25","checkIn":"<REQUIRED>","checkOut":"<REQUIRED>","adults1":"1","sortOrder":"PRICE","locale":"en_US","currency":"USD"}

headers = {
	"X-RapidAPI-Key": "Ваш ключ",
	"X-RapidAPI-Host": "hotels4.p.rapidapi.com"
}

response = requests.request("GET", url, headers=headers, params=querystring)

print(response.text)
